src/Nodes1DProvisioner.cpp

```diff
@@ -80,6 +86,23 @@ void Nodes1DProvisioner::buildNodes() {
 
     buildConnectivityMatrices();
     buildFaceMask();
+	buildMaps();
+    buildNormals();
+}
+
+/**
+ * Build unit normals at element faces. Trivial in 1D.
+ */
+void Nodes1DProvisioner::buildNormals() {
+    matrix_type & nxref = *nx;
+
+    index_type count = 0;
+    for (index_type k=0; k < NumElements; k++) {
+        for (index_type f=0; f < NumFaces*NumFacePoints; f++) {
+            nxref(f,k) = pow(-1., count+1);
+            count++;
+        }
```

Comment:
Hey bud, I know you love the pow function ;-), but I would implement this as

real_type mult = -1;

and then inside the function:

nxref(f, k) = count;
count *= -1;

Reply:
The pow function is the greatest, but I will abide by your advice, lol.